packages/template-retail-react-app/app/pages/cart/index.jsx

```diff
@@ -448,6 +456,8 @@ const Cart = () => {
                                                     ...productItem,
                                                     ...(products &&
                                                         products[productItem.productId]),
+                                                    isProductUnavailable:
+                                                        !products?.[productItem.productId],
```

Comment:
This currently results in the flash of Unavailable badge showing briefly, before the product image is rendered. Can we tweak this logic to avoid the flash?

How to reproduce the issue:
1. Navigate to the Cart page
2. Observe the product image for the Unavailable badge
3. Reload the page to observe it again

Reply:
fixed.